### PR TITLE
add endpoint to create main service account

### LINKED ISSUE #6644: Add V2 endpoint to create main SA

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: add an endpoint to create a main service account.
Refactor old provider methods from service account to project service account.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6644

```release-note
Service Accounts V2
```

### CODE REVIEW

File: pkg/handler/v2/serviceaccount/sa.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
```
Comment (lsviben) [inline pkg/handler/v2/serviceaccount/sa.go]: NIT: 2021

File: pkg/handler/v2/serviceaccount/sa.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	kubermaticapiv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/handler/v1/common"
+	serviceaccount "k8c.io/kubermatic/v2/pkg/provider/kubernetes"
```
Comment (lsviben) [inline pkg/handler/v2/serviceaccount/sa.go]: NIT: reorganize imports

File: pkg/provider/types.go [hunk 3]
```diff
@@ -520,45 +520,48 @@ func DatacenterCloudProviderName(spec *kubermaticv1.DatacenterSpec) (string, err
...
+	DeleteProjectServiceAccount(userInfo *UserInfo, name string) error
+
+	ListMainServiceAccounts(userInfo *UserInfo, options *ServiceAccountListOptions) ([]*kubermaticv1.User, error)
+	CreateMainServiceAccount(userInfo *UserInfo, name, group string) (*kubermaticv1.User, error)
 }
 
 // PrivilegedServiceAccountProvider declares the set of methods for interacting with kubermatic service account
 type PrivilegedServiceAccountProvider interface {
-	// CreateUnsecured creates a service accounts
+	// CreateUnsecuredProjectServiceAccount creates a service accounts
```
Comment (lsviben) [inline pkg/provider/types.go]: creates a project service account

File: pkg/provider/types.go [hunk 4]
```diff
@@ -520,45 +520,48 @@ func DatacenterCloudProviderName(spec *kubermaticv1.DatacenterSpec) (string, err
...
 	//
 	// Note that this function:
 	// is unsafe in a sense that it uses privileged account to create the resources
-	CreateUnsecured(project *kubermaticv1.Project, name, group string) (*kubermaticv1.User, error)
+	CreateUnsecuredProjectServiceAccount(project *kubermaticv1.Project, name, group string) (*kubermaticv1.User, error)
 
-	// ListUnsecured gets all service accounts
+	// ListUnsecuredProjectServiceAccount gets all service accounts
```
Comment (lsviben) [inline pkg/provider/types.go]: gets all project service accounts

File: pkg/provider/types.go [hunk 5]
```diff
@@ -520,45 +520,48 @@ func DatacenterCloudProviderName(spec *kubermaticv1.DatacenterSpec) (string, err
...
 	// If you want to filter the result please take a look at ServiceAccountListOptions
 	//
 	// Note that this function:
 	// is unsafe in a sense that it uses privileged account to get the resources
-	ListUnsecured(project *kubermaticv1.Project, options *ServiceAccountListOptions) ([]*kubermaticv1.User, error)
+	ListUnsecuredProjectServiceAccount(project *kubermaticv1.Project, options *ServiceAccountListOptions) ([]*kubermaticv1.User, error)
 
-	// GetUnsecured gets all service accounts
+	// GetUnsecuredProjectServiceAccount gets all service accounts
```
Comment (lsviben) [inline pkg/provider/types.go]: gets a project service account

File: pkg/provider/types.go [hunk 6]
```diff
@@ -520,45 +520,48 @@ func DatacenterCloudProviderName(spec *kubermaticv1.DatacenterSpec) (string, err
...
 	//
 	// Note that this function:
 	// is unsafe in a sense that it uses privileged account to get the resource
-	GetUnsecured(name string, options *ServiceAccountGetOptions) (*kubermaticv1.User, error)
+	GetUnsecuredProjectServiceAccount(name string, options *ServiceAccountGetOptions) (*kubermaticv1.User, error)
 
-	// UpdateUnsecured gets all service accounts
+	// UpdateUnsecuredProjectServiceAccount gets all project service accounts
```
Comment (lsviben) [inline pkg/provider/types.go]: updates a project service account

**Review by lsviben — CHANGES_REQUESTED** [review]
Looks ok, some minor comments

**Comment by zreigz** [comment]
@lsviben done PTAL

**Pull request merged**